### PR TITLE
Disable cache emulation checks for VC

### PR DESCRIPTION
**Pull request by aglab2**

### CODE REVIEW

File: src/boot/main.c [hunk 1]
```diff
@@ -346,7 +346,11 @@ void thread3_main(UNUSED void *arg) {
         gIsConsole = FALSE;
         gBorderHeight = BORDER_HEIGHT_EMULATOR;
         gIsVC = IS_VC();
```
Comment (gheskett) [inline src/boot/main.c:348]: Can this or the one at the top be removed? Not sure why this is getting called twice.

**Comment by gheskett** [comment]
Another question, does RCVI Hack break VC? If so, I'll need to throw in another check for it in the PR.

File: src/boot/main.c (hunk 1 above)
Comment (aglab2) [reply src/boot/main.c]: Got rid of unnecessary gIsVC. if I rememeber right, it was done early to allow using gIsVC in EEPROM setup so the one inside `IO_READ(DPC_CLOCK_REG) == 0` must be unnecessary.

**Comment by aglab2** [comment]
> 

RCVI hack does not break WiiVC. I think it just straight out ignored and VC is pumping as much performance as it can without considering N64 cycles. I tested it on Serene Fusion and it worked just fine (except for random VC crashes which is something I need to look into).

**Review by gheskett — APPROVED** [review]
I don't honestly know all that much about the Wii's runtime; I assume VC would struggle with `BETTER_REVERB` actually which is something I've forgotten to consider, but that's a discussion for another PR I suppose. Perhaps a `gIsConsoleOrVC` variable is warranted eventually?

**Comment by arthurtilly** [comment]
use an && not a new variable, you disgust me...

**Comment by aglab2** [comment]
Using && in this case will make it less clear what the intention is - not to launch code on a particular platform. Moreover there might be more checks that will break WiiVC but not N64/emus, will be easier to expand.

**Comment by arthurtilly** [comment]
well if a check breaks wii vc but not n64 then gIsConsoleOrVC doesnt work for it anyway lol. also i meant || - just `gIsConsole || gIsVC` is perfectly clear, a new variable is completely unnecessary

**Comment by aglab2** [comment]
If you mean `gCacheEmulated`, it is not my doing :) I'd rather keep it for clarity for future generations that will try to figure out what magic was done in HackerSM64

**Comment by arthurtilly** [comment]
oh maybe there is confusion, i was not commenting on the pr, i was commenting on arctic's "Perhaps a gIsConsoleOrVC variable is warranted eventually?" comment earlier

**Comment by gheskett** [comment]
> use an && not a new variable, you disgust me...

Eh, sure, but as it stands, I count at least 6 locations where it could come in handy (which would actually exceed uses of gIsConsole standalone). Not a big deal either way though.

**Comment by aglab2** [comment]
I think the best solution to solve the variable hell would be using `gPlatform` enum with values CONSOLE, VC, EMU and ACCURATE_EMU. I can expect any of these platforms to require particular patches.

**Pull request merged**